packages/snaps-simulator/src/features/builder/utils.ts

```diff
@@ -105,82 +196,75 @@ function getComponentTypes(component: Panel | Form): NodeType[] {
 }
 
 /**
- * Get the arguments used to create a component.
+ * Convert an element to code.
  *
- * @param component - The component.
- * @returns The arguments.
+ * @param element - The element.
+ * @returns The element as code.
  */
-function getComponentArgs(component: Component): string {
-  switch (component.type) {
-    case NodeType.Panel:
-      return component.children.map(getComponentArgs).join(',\n');
-    case NodeType.Form:
-      return `'${component.name}', ${component.children
-        .map(getComponentArgs)
-        .join(',\n')}`;
-    case NodeType.Copyable:
-      return `'${component.value}'${component.sensitive ? ', true' : ''}`;
-    case NodeType.Text:
-    case NodeType.Heading:
-    case NodeType.Image:
-      return JSON.stringify(component.value);
-    case NodeType.Button:
-    case NodeType.Input: {
-      const args = Object.keys(component)
-        .filter(
-          (key) =>
-            key !== 'type' && component[key as keyof typeof component] !== '',
-        )
-        .reduce((acc, prev) => {
-          return { ...acc, [prev]: component[prev as keyof typeof component] };
-        }, {});
-
-      return JSON.stringify(args);
-    }
-    case NodeType.Spinner:
-    case NodeType.Divider:
-    default:
-      return '';
+function elementToCode(element: JSXElement): string {
+  const children = getJsxChildren(element);
+
+  if (children.length === 0) {
+    return `<${element.type}${getElementProps(element.props)} />`;
   }
+
+  return `<${element.type}${getElementProps(
+    element.props,
+  )}>${`${getElementChildren(getJsxChildren(element))}</${element.type}>`}`;
 }
 
 /**
- * Get the code for a component.
+ * Get the props of an element as code.
  *
- * @param component - The component.
- * @returns The code.
+ * @param props - The props.
+ * @returns The props to code.
  */
-function componentToCode(component: Component): string {
-  if (component.type === NodeType.Panel) {
-    return `panel([\n${component.children
-      .map(componentToCode)
-      .join(',\n')}\n])`;
-  }
+function getElementProps(props: Record<string, unknown>) {
```

Comment:
We may be able to share some code between this and `snaps-jest` here.

Reply:
Can you point me to what part you were thinking we could reuse ? I've searched in `snaps-jest` but nothing seems relevant.

Reply:
I mean this part:

https://github.com/MetaMask/snaps/blob/3866525b483f3ca33ed68398292d2c3f5e65fb84/packages/snaps-jest/src/matchers.ts#L187-L248

It turns a rendered JSX node into code.